packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -856,35 +860,64 @@ function simplifyDispatcher<T>(
 		context.ctx.logger.warn(`Tried to access unknown dispatcher ${typeDef.registry}`)
 		return { kind: 'union', members: [] }
 	}
-	const structFields: StructTypePairField[] = []
-	for (const key in dispatcher) {
-		const data = dispatcher[key].data
-		if (TypeDefSymbolData.is(data)) {
-			structFields.push({ kind: 'pair', key, type: data.typeDef })
-		}
-	}
-	return simplifyIndexed({
-		kind: 'indexed',
-		parallelIndices: typeDef.parallelIndices,
-		child: { kind: 'struct', fields: structFields },
-	}, context)
+	return resolveIndices<T>(typeDef.parallelIndices, dispatcher, context)
 }
 
 function simplifyIndexed<T>(
 	typeDef: IndexedType,
 	context: SimplifyContext<T>,
 ): SimplifiedMcdocType {
-	const child = simplify(typeDef.child, { ...context, typeArgs: [] })
+	const child = simplify(typeDef.child, {
+		...context,
+		typeArgs: [],
+		structFields: undefined,
+	})
 
 	if (child.kind !== 'struct') {
 		context.ctx.logger.warn(`Tried to index un-indexable type ${child.kind}`)
 		return { kind: 'union', members: [] }
 	}
+
+	const symbolMap: { [key: string]: { data?: unknown } } = {}
+	for (const field of child.fields) {
+		if (field.key.kind === 'literal' && field.key.value.kind === 'string') {
+			symbolMap[field.key.value.value] = {
+				data: {
+					typeDef: field.type,
+				} satisfies TypeDefSymbolData,
+			}
+		}
+	}
+
+	return resolveIndices(typeDef.parallelIndices, symbolMap, context)
+}
+
+function resolveIndices<T>(
+	parallelIndices: ParallelIndices,
+	symbolMap: { [key: string]: { data?: unknown } },
+	context: SimplifyContext<T>,
+): SimplifiedMcdocType {
 	let values: McdocType[] = []
 
-	for (const index of typeDef.parallelIndices) {
+	let unkownTypeDef: McdocType | undefined | false = false
+	function getUnknownTypeDef() {
+		if (unkownTypeDef === false) {
+			const data = symbolMap['%unknown']?.data
+			unkownTypeDef = TypeDefSymbolData.is(data) ? data.typeDef : undefined
+		}
+		return unkownTypeDef
+	}
+
+	for (const index of parallelIndices) {
 		let lookup: string[] = []
 		if (index.kind === 'static') {
+			if (index.value === '%fallback') {
```

Comment:
Yes this is precisely why I re-added it.

This code will run when explicitly trying to access %fallback. Without this, %fallback would act like %unknown, as it would try to find an explicitly dispatched fallback, not find anything, and then run the %unknown case.

So this was correct before, I just forgot about %unknown when I was in voice with Mulv.